### PR TITLE
Hapi/Seneca API refactor

### PR DESCRIPTION
**Pull request by dberesford**
* API is now all in Hapi routes
* Removed a lot of Seneca, only using seneca client now from Hapi routes (no seneca-web)
* API refactor, api is now version 2.0
* Misc smaller refactorings

### CODE REVIEW

File: lib/dojos.js [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+    method: 'POST',
+    path: options.basePath + '/dojos/save-dojo-lead',
+    handler: handlers.actHandlerNeedsUser('save_dojo_lead')
+  }, {
+    method: 'PUT',
+    path: options.basePath + '/dojos/update-dojo-lead/{id}',
+    handler: handlers.actHandlerNeedsUser('update_dojo_lead', 'id')
+  }, {
+    method: 'GET',
+    path: options.basePath + '/dojos/user-dojo=lead/{id}',
```
Comment (ckiss) [inline lib/dojos.js]: typo here user-dojo=lead

File: lib/dojos.js [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+    path: options.basePath + '/dojos/state-count/{country}',
+    handler: handlers.actHandler('dojos_state_count', 'country')
+  }, {
+    method: 'POST',
+    path: options.basePath + '/dojos/list',
+    handler: handlers.actHandler('list')
+  }, {
+    method: 'POST',
+    path: options.basePath + '/dojos',
+    handler: handlers.actHandler('list')
```
Comment (ckiss) [inline lib/dojos.js]: this is duplicated

File: lib/dojos.js [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+      }
+    }
+  }, {
+    method: 'POST',
+    path: options.basePath + '/countries/places',
+    handler: handlers.actHandler('list_places')
+  }, {
+    method: 'GET',
+    path: options.basePath + '/countries/continents/lat-long',
+    handler: handlers.actHandler('continents_lat_long')
```
Comment (ckiss) [inline lib/dojos.js]: shouldn't have `expiresIn` as `list_countries` above? It did had in old version of code.

File: lib/dojos.js [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+  }, {
+    method: 'GET',
+    path: options.basePath + '/countries/lat-long',
+    handler: handlers.actHandler('countries_lat_long')
```
Comment (ckiss) [inline lib/dojos.js]: same as above

File: lib/dojos.js [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+  }, {
+    method: 'GET',
+    path: options.basePath + '/countries/continents/codes',
+    handler: handlers.actHandler('continent_codes')
```
Comment (ckiss) [inline lib/dojos.js]: same here :)

File: lib/dojos.js [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+  }, {
+    method: 'GET',
+    path: options.basePath + '/countries/continents',
+    handler: handlers.actHandler('countries_continents')
```
Comment (ckiss) [inline lib/dojos.js]: same here :)

File: lib/dojos.js [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+  }]);
+
+  next();
+};
+
+exports.register.attributes = {
+  name: 'api-dojos'
+};
```
Comment (ckiss) [inline lib/dojos.js]: also I've seen dojos_count is not present anymore. Is that intended because it's not used anymore?

File: lib/events.js [hunk 8]
```diff
@@ -0,0 +1,207 @@
...
+    method: 'POST',
+    path: options.basePath + '/events/save',
+    handler: validateEvent('saveEvent')
+  }, {
+    method: 'GET',
+    path: options.basePath + '/events/{id}',
+    handler: handlers.actHandler('getEvent', 'id')
+  }, {
+    method: 'POST',
+    path: options.basePath + '/listEvents',
```
Comment (ckiss) [inline lib/events.js]: the other way around: path should be `/events` and handler should be `listEvents`

File: lib/profiles.js [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+    path: options.basePath + '/profiles/youth/update',
+    handler: handlers.actHandlerNeedsUser('update-youth-profile')
+  }, {
+    method: 'POST',
+    path: options.basePath + '/profiles/invite-parent-guardian',
+    handler: handlers.actHandlerNeedsUser('invite-parent-guardian')
+  }, {
+    method: 'POST',
+    path: options.basePath + '/profiles/accept-parent-guardian',
+    handler: handlers.actHandlerNeedsUser('accept-parent-invite')
```
Comment (ckiss) [inline lib/profiles.js]: low dash here

File: lib/profiles.js [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+    path: options.basePath + '/profiles/{id}/avatar',
+    handler: handlers.actHandler('get_avatar', 'id')
+  }, {
+    method: 'GET',
+    path: options.basePath + '/profiles/{id}/avatar_img',
+    handler: handleAvatarImage
+  }, {
+    method: 'GET',
+    path: options.basePath + '/profiles/{id}',
+    handler: handlers.actHandlerNeedsUser('get_avatar', 'id')
```
Comment (ckiss) [inline lib/profiles.js]: duplicated (see line 50)

File: lib/users.js [hunk 11]
```diff
@@ -0,0 +1,151 @@
...
+    method: 'POST',
+    path: options.basePath + '/logout',
+    handler: handleLogout
+  }, {
+    method: 'GET',
+    path: options.basePath + '/instance',
+    handler: handleInstance
+  }, {
+    method: 'POST',
+    path: options.basePath + '/list',
```
Comment (ckiss) [inline lib/users.js]: users list? do we still need this?

File: lib/users.js [hunk 12]
```diff
@@ -0,0 +1,151 @@
...
+    path: options.basePath + '/dojo-admins-for-user/{userId}',
+    handler: handlers.actHandler('load_dojo_admins_for_user', 'userId')
+  }, {
+    method: 'GET',
+    path: options.basePath + '/kpi/number-of-youths-registered',
+    handler: handlers.actHandlerNeedsCdfAdmin('kpi_number_of_youths_registered')
+  }, {
+    method: 'GET',
+    path: options.basePath + '/kpi/number-of-champions-and-mentors-registered',
+    handler: handlers.actHandlerNeedsCdfAdmin('number_of_champions_and_mentors_registered')
```
Comment (ckiss) [inline lib/users.js]: shouldn't handler remain `kpi_number_of_champions_and_mentors_registered`?

File: lib/users.js [hunk 13]
```diff
@@ -0,0 +1,151 @@
...
+  }, {
+    method: 'GET',
+    path: options.basePath + '/kpi/number-of-youth-females-registered',
+    handler: handlers.actHandlerNeedsCdfAdmin('number_of_youth_females_registered')
```
Comment (ckiss) [inline lib/users.js]: same as above `kpi_number_of_youth_females_registered`

**Comment by ckiss** [comment]
@dberesford some comments here and there, mainly typos, but otherwise LGTM

File: lib/dojos.js (hunk 7 above)
Comment (dberesford) [inline lib/dojos.js]: yes not used

File: lib/profiles.js (hunk 9 above)
Comment (dberesford) [inline lib/profiles.js]: sorry @ckiss what do you mean here exactly? i.e. 'low dash'

Comment (ckiss) [inline lib/profiles.js]: sorry @dberesford I meant handler names should use underscore, like accept_parent_invite :)

**Comment by dberesford** [comment]
Awesome code review @ckiss - thanks, much appreciated!

File: lib/profiles.js (hunk 9 above)
Comment (dberesford) [inline lib/profiles.js]: ah - no, that's what they are, it's very inconsistent

**Comment by ckiss** [comment]
thanks @dberesford :) no worries at all :)

**Pull request merged**